text/0000-rfc-release-notes.md

```diff
@@ -0,0 +1,82 @@
+# Meta
+[meta]: #meta
+- Name: Additions to RFC template
+- Start Date: 4/7/2021
+- Author(s): natalieparellano
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+The existing process for creating text for release notes is largely manual. This RFC proposes adding new fields to the RFC template that could be parsed by (future) automation and added to issues and ultimately release notes via more automation.
+
+# Motivation
+[motivation]: #motivation
+
+- Why should we do this? Automate a manual process that is currently being done by contributors and maintainers.
+- What use cases does it support? Release notes for: spec, lifecycle, pack, etc.
+- What is the expected outcome? More consistency in release notes text and less time spent on this process.
+
+# What it is
+[what-it-is]: #what-it-is
+
+- Define the target persona: project contributor.
+
+Example: For opt-in layer caching, we had the following release notes:
+
+* spec: `Buildpacks must explicitly opt-in to layer re-use by appending to <layers>/<layer>.toml (#132, #209, RFC 0052).`
+* lifecycle: `When using buildpack API 0.6 or greater, buildpacks must explicitly opt-in to layer re-use by appending to <layers>/<layer>.toml (#537 by @yaelharel)`
+* migration guide: 
+```text
+Buildpacks must now explicitly opt-in to layer re-use by setting the launch, build, and cache keys in <layers>/<layer>.toml. If buildpacks do not modify <layers>/<layer>.toml, the layer will behave like a temporary directory, available only to the authoring buildpack, existing for the duration of a single build - even if the buildpack in the previous build set any of these keys to true.
+
+NOTE: Additionally, the launch, build, and cache keys are moved under a new [types] table in <layers>/<layer>.toml and removed from the top level.
+
+A Bash buildpack could write something like the following to <layers>/<layer>.toml in order to cache a layer:
+
+cat >> layer.toml <<EOL
+[types]
+cache = true
+EOL
+```
+
+It would be nice if this text was provided by the RFC author in the RFC template, so that when issues are generated from the RFC, the issues would already contain this text. Then when release notes are generated from those issues, the release notes would already contain this text. We could even automate the creation of the migration guides.
+
+Example: In the RFC template:
+```text
+# Release Notes (OPTIONAL)
+Provide a one-sentence summary of the feature that could be used to announce this feature to buildpacks users.
```

Comment:
Lot's of RFCs touch multiple components which might need slightly different release notes. For example if an RFC add support for platform provided labels to the lifecycle and `pack` allows those users to set those labels with a flag you might want:
"Allows platform to provide additional image labels"
vs.
"Allows users to set labels during `pack build` with the `--label` flag
vs.
"Supports labels in project.toml`

Most release note autogeneration uses the issue or PR title generate release notes. So maybe https://github.com/buildpacks/rfcs/blob/rfc-issue-generation/text/0000-rfc-issue-generation.md already gives us a way to automate this in a per-component way?

Reply:
@ekcasey are you suggesting to do this using comments after the fact?

Reply:
Maybe this is something that could be solved by editing generated issue titles after the fact.

If https://github.com/buildpacks/rfcs/blob/rfc-issue-generation/text/0000-rfc-issue-generation.md uses PR title to generate the issue title, then issues would have names like:

* [RFC 0073] Offline Buildpackages RFC
* [RFC 0075] Move analyze phase before detect

which aren't that great for generating release notes, but the maintainers of the component repos could then edit the issue titles so that they are more useful.

I was originally thinking about asking RFC authors to choose more descriptive PR titles, but maybe it's too much. It might be overly cumbersome when writing an RFC.